python/pacemaker/_cts/network.py

```diff
@@ -0,0 +1,34 @@
+""" Network related utilities for CTS """
+
+__all__ = ["next_ip"]
+__copyright__ = "Copyright 2000-2023 the Pacemaker project contributors"
+__license__ = "GNU General Public License version 2 or later (GPLv2+) WITHOUT ANY WARRANTY"
+
+def next_ip(env):
+    """ Return the next available IP address """
+
+    ip = env["IPBase"]
+
+    new_ip = None
+
+    # Split the existing IP address up into a tuple of:
+    # (everything except the last part of the addr, the separator, the last part of the addr).
+    # For instance, "192.168.1.2" becomes ("192.168.1", ".", "2").  Then,
+    # increment the last part of the address and paste everything back
+    # together.
+    if ":" in ip:
+        # This is an IPv6 address
+        fields = ip.rpartition(":")
+        new_ip = hex(int(fields[2], 16) + 1)
+
+    else:
+        # This is an IPv4 address
+        fields = ip.rpartition(".")
+        new_ip = int(fields[2]) + 1
+
+    if new_ip > 255:
+        raise ValueError("No more available IP addresses")
+
+    ip = "%s%s%s" % (fields[0], fields[1], new_ip)
+    env["IPBase"] = ip
```

Comment:
Eventually, we should avoid modifying the initial IPBase value

Reply:
It'd be pretty easy to use a global that just doesn't get exported and update that.  I'm sure there's something more elegant that could be done, but I'm not sure it's worth the effort.